src/transformers/modeling_utils.py

```diff
@@ -1768,7 +1768,8 @@ def save_pretrained(
             # We're going to remove aliases before saving
             ptrs = collections.defaultdict(list)
             for name, tensor in state_dict.items():
-                ptrs[tensor.data_ptr()].append(name)
+                ident = (tensor.data_ptr(), tensor.device, tensor.shape, tensor.stride())
```

Comment:
This change exists because of multi-gpu setup and potential peculiar sharing of tensors.

Tensors are considered shared, and droppable if and only if they are the exact same tensor.
same ptr, same device, same shape, same stride.

We don't need to handle device meta here I think since trying to save a model with device meta should already be a bug.